nucleus/dataset.py

```diff
@@ -199,16 +200,16 @@ def ingest_tasks(self, task_ids: dict):
 
     def append(
```

Comment:
What will happen if you try to append dataset items to a 3D dataset? I thought we were going to ban this

Reply:
this requires changes on the scaleapi side, we currently do not track the notion of 3d vs. 2d dataset in postgres.  I'd recommend considering this out of scope for this particular PR

Reply:
will need to add dataset validation for this in follow up PR